packages/teams-js/src/public/videoEffects.ts

```diff
@@ -1,12 +1,18 @@
-import { sendMessageToParent } from '../internal/communication';
-import { registerHandler } from '../internal/handlers';
+import { sendMessageToParent, sendMessageToParentWithVersion } from '../internal/communication';
+import { registerHandlerWithVersion } from '../internal/handlers';
 import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
 import { inServerSideRenderingEnvironment, ssrSafeWindow } from '../internal/utils';
 import { createEffectParameterChangeCallback, processMediaStream } from '../internal/videoEffectsUtils';
 import { VideoPerformanceMonitor } from '../internal/videoPerformanceMonitor';
 import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
 import { runtime } from './runtime';
 
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const videoEffectsTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
 /**
  * Namespace to video extensibility of the SDK
  * @beta
```

Comment:
@TrevorJoelHarris, I have a question here.

If you take a look on line 23, it seems like sendMessageToParent is passed to VideoPerformanceMonitor Class to initialize an Object, which goes to file (packages/teams-js/src/internal/videoPerformanceMonitor.ts). And under that constructor, it continuously sends out request.

What is request 'video.performance.performanceDataGenerated' used for? Should I treat it as an API to assign an API version tag for it? I think so?

Reply:
You'd have to ask the author to be sure, but I believe this is to send data up to the host semi-continuously to make sure they have a good handle on the framerate the video effects are running at.

Whether you want telemetry for this or not is a good question since there won't be a 1:1 relationship between 'developer called a function' and 'this telemetry point was sent up`. Maybe worth tracking down the author, making sure you know what this is for, and then talking with PM about what they would like you to do here?

Reply:
@KangxuanYe followed up with the owner and after that discussion feels good that this is the correct way to telemetrize this function (is telemetrize a word?)